build-system/pr-check/utils.js

```diff
@@ -0,0 +1,141 @@
+      'https://amphtml-sauce-token-dealer.appspot.com/getJwtToken').trim();
+  const startScCmd = 'build-system/sauce_connect/start_sauce_connect.sh';
+  const fileLogPrefix = colors.bold(colors.yellow(`${functionName}:`));
+  console.log('\n' + fileLogPrefix,
+      'Starting Sauce Connect Proxy:', colors.cyan(startScCmd));
+  execOrDie(startScCmd);
+}
+
+/**
+ * Stops connection to Sauce Labs
+ * @param {string} functionName
+ */
+function stopSauceConnect(functionName) {
+  const stopScCmd = 'build-system/sauce_connect/stop_sauce_connect.sh';
+  const fileLogPrefix = colors.bold(colors.yellow(`${functionName}:`));
+  console.log('\n' + fileLogPrefix,
+      'Stopping Sauce Connect Proxy:', colors.cyan(stopScCmd));
+  execOrDie(stopScCmd);
+}
+
+/**
+ * Starts a timer to measure the execution time of the given function.
+ * @param {string} functionName
+ * @param {string} fileName
+ * @return {DOMHighResTimeStamp}
+ */
+function startTimer(functionName, fileName) {
+  const startTime = Date.now();
+  const fileLogPrefix = colors.bold(colors.yellow(`${fileName}:`));
+  console.log(
+      '\n' + fileLogPrefix, 'Running', colors.cyan(functionName) + '...');
+  return startTime;
+}
+
+/**
+ * Stops the timer for the given function and prints the execution time.
+ * @param {string} functionName
+ * @param {DOMHighResTimeStamp} startTime
+ * @return {number}
+ */
+function stopTimer(functionName, startTime) {
+  const endTime = Date.now();
+  const executionTime = new Date(endTime - startTime);
+  const mins = executionTime.getMinutes();
+  const secs = executionTime.getSeconds();
+  const fileLogPrefix = colors.bold(colors.yellow(`${functionName}:`));
+  console.log(
+      fileLogPrefix, 'Done running', colors.cyan(functionName),
+      'Total time:', colors.green(mins + 'm ' + secs + 's'));
+}
+
+/**
+ * Executes the provided command and times it. Errors, if any, are printed.
+ * @param {string} cmd
+ * @return {<Object>} Process info.
+ */
+function timedExec(cmd) {
+  const startTime = startTimer(cmd);
+  const p = exec(cmd);
+  stopTimer(cmd, startTime);
+  return p;
+}
+
+/**
+ * Executes the provided command and times it. The program terminates in case of
+ * failure.
+ * @param {string} cmd
+ * @param {string} fileName
+ */
+function timedExecOrDie(cmd, fileName = 'utils.js') {
+  const startTime = startTimer(cmd, fileName);
+  execOrDie(cmd);
+  stopTimer(fileName, startTime);
+}
+
+
+function unzipBuildOutput() {
+  timedExecOrDie('gsutil cp ' +
+      `${BUILD_OUTPUT_STORAGE_LOCATION}/${BUILD_OUTPUT_FILE} ` +
+      `${BUILD_OUTPUT_FILE}`);
+  timedExecOrDie(
+      `log="$(unzip -o ${BUILD_OUTPUT_FILE})" && ` +
+      'echo travis_fold:start:unzip_results && echo ${log} && ' +
+      'echo travis_fold:end:unzip_results');
+  timedExecOrDie(
+      `log="$(ls -la ${BUILD_OUTPUT_DIRS})" && ` +
+      'echo travis_fold:start:verify_unzip_results && echo ${log} && ' +
+      'echo travis_fold:end:verify_unzip_results');
```

Comment:
There's no need to time the zip operations here, since the output looks weird. Just call `execOrDie` from `exec.js` and apply the log folding above and below the unzip commands.

Reply:
Alternative implementation:

```
function downloadBuildOutput() {
  const buildOutputDownloadUrl = `${BUILD_OUTPUT_STORAGE_LOCATION}/${BUILD_OUTPUT_FILE}`;
  console.log(fileLogPrefix,
      'Downloading build output from',
      colors.cyan(buildOutputDownloadUrl) + '...');
  execOrDie(`gsutil cp ${buildOutputDownloadUrl} ${BUILD_OUTPUT_FILE}`);
  console.log(fileLogPrefix,
      'Extracting',
      colors.cyan(BUILD_OUTPUT_FILE) + '...');
  exec('echo travis_fold:start:unzip_results');
  execOrDie(`unzip -o ${BUILD_OUTPUT_FILE}`);
  exec('echo travis_fold:end:unzip_results');
  console.log(fileLogPrefix, 'Verifying extracted files...');
  exec('echo travis_fold:start:verify_unzip_results');
  execOrDie(`ls -la ${BUILD_OUTPUT_DIRS}`);
  exec('echo travis_fold:end:verify_unzip_results');
}
```

Reply:
Done!